### PR TITLE
Improve dockerfile

### PR DESCRIPTION
**Pull request by fredericfran-gds**
⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -0,0 +1,8 @@
+.git
+.gitignore
+Jenkinsfile
+README.md
+docs
+log
+spec
+tmp
```
Comment (nsabri1) [inline .dockerignore:8]: Are you sure these directories are not needed - have you tested your changes against E2E checks - are your sure spec folder is not needed part of the build.

File: Dockerfile [hunk 2]
```diff
@@ -1,22 +1,36 @@
+# TODO: make this default to govuk-ruby once it's being pushed somewhere public
 ARG base_image=ruby:2.7.2
-FROM ${base_image}
+ARG app_home=/app
 
-RUN apt-get update -qq && apt-get upgrade -y
```
Comment (nsabri1) [inline Dockerfile:4]: Is there reason why you are not updating the image?
same questions about installing build-essentials and Nodejs packages? are they needed for this app?

File: Dockerfile [hunk 3]
```diff
@@ -1,22 +1,36 @@
...
+
+RUN bundle config set deployment 'true' && \
+    bundle config set without 'development test' && \
+    bundle install -j8 --retry=2
+
+COPY . ${app_home}
+
+
+FROM $base_image
+
```
Comment (nsabri1) [inline Dockerfile:25]: This image should be updated and patched with apt-get update and upgrade

File: .dockerignore (hunk 1 above)
Comment (fredericfran-gds) [reply .dockerignore:8]: I took it originally from publisher and modify the list accordingly: https://github.com/alphagov/publisher/blob/main/.dockerignore

File: Dockerfile (hunk 2 above)
Comment (fredericfran-gds) [reply Dockerfile:4]: I added them again but the packages are only updated only at build time so we will need to rebuild frequently to get the updates.

File: Dockerfile (hunk 3 above)
Comment (fredericfran-gds) [reply Dockerfile:28]: I added them again but the packages are only updated only at build time so we will need to rebuild frequently to get the updates.

**Pull request closed without merging**